src/odemis/driver/test/technolution_test.py

```diff
@@ -432,6 +428,95 @@ def test_resolution_VA(self):
         self.EBeamScanner.resolution.value = (6385, 6385)
         self.assertEqual(self.EBeamScanner.resolution.value, (6400, 6400))
 
+    def test_VA_dependencies(self):
+        """ Testing the dependencies between the scanner.resolution, mppc.cellCompleteResolution and
+        mppc.cellTranslation VAs. """
+
+        # Set of values that fulfil the rule: cell complete size >= effective cell size + max cell translation
+        # However, a certain order to set these values is required, as the setters of the VAs do not adjust
+        # values automatically yet. The following examples demonstrate the order the VAs need to be adjusted in
+        # order to not raise a value error.
+
+        # initial values
+        self.MPPC.cellCompleteResolution.value = (900, 900)
+        self.EBeamScanner.resolution.value = (6400, 6400)
+        self.MPPC.cellTranslation.value = \
+            tuple(tuple((50, 50) for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
+
+        # First example increase sizes: first cellCompleteResolution, then order independent
+        # (1000, 1000) >= (910, 910) + (80, 80)
+        self.MPPC.cellCompleteResolution.value = (1000, 1000)
+        self.EBeamScanner.resolution.value = (7280, 7280)  # -> eff cell size = (910, 910)
+        self.MPPC.cellTranslation.value = \
+            tuple(tuple((80, 80) for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
+
+        # Second example decreases sizes: first resolution and cellTranslation, then cellCompleteResolution
+        # (800, 800) + (50, 50) <= (850, 850)
+        self.EBeamScanner.resolution.value = (6400, 6400)  # -> eff cell size = (800, 800)
+        self.MPPC.cellTranslation.value = \
+            tuple(tuple((50, 50) for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
+        self.MPPC.cellCompleteResolution.value = (850, 850)
+
+        # Third example increase sizes 2:
+        # first cellCompleteResolution, then reduce cellTranslation, then increase resolution
+        # (1000, 1000) >= (980, 980) + (20, 20)
+        self.MPPC.cellCompleteResolution.value = (1000, 1000)
+        self.MPPC.cellTranslation.value = \
+            tuple(tuple((20, 20) for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
+        self.EBeamScanner.resolution.value = (7840, 7840)  # -> eff cell size = (980, 980)
```

Comment:
The more I'm looking at it, the less I think it's going to be easy to write reliable client code. What is the rule? Are you able to write a function which takes are arguments resolution, cell translation, and cell complete resolution, and always succeeds in writing it?

Let's say it's currently at 7280, 1000 , 80 (like example 1), and you want to do example 3: 7840, 1000, 20. If you do it in the order of example 3, it works. But if you do in the order of example 1, it fails when attempting to set the resolution (because it'd see 1000 < 980 + 80).

Even more difficult is the case where things are not square (and admittedly, a little unlikely to happen in reality). Imagine it starts at 6400x7280, 900x1000, 50x80, and want to go to 7280x6400, 1000x900, 80x50. 
Do you need to follow the code that increase or decrease? I guess the only way would be to use intermediary values. Maybe first set the cell complete resolution to the max, then change the rest, and then apply the requested cell complete resolution.